lualib/rspamadm/configwizard.lua

```diff
@@ -406,22 +409,12 @@ local function setup_dkim_signing(cfg, changes)
     if not rspamd_util.file_exists(privkey_file) then
       if ask_yes_no("Do you want to create privkey " .. highlight(privkey_file),
           true) then
-        local pubkey_file = privkey_file .. ".pub"
-        local rspamd_cryptobox = require "rspamd_cryptobox"
-        local sk, pk = rspamd_cryptobox.generate_keypair("rsa", 2048)
-        pk:save_to_file(pubkey_file)
-        sk:save_to_file(privkey_file, tonumber('0600', 8))
-
-        local f = io.open(pubkey_file)
-        if not f then
-          printf("Cannot open pubkey file %s, fatal error", highlight(pubkey_file))
-          os.exit(1)
-        end
-
-        local content = f:read("*all")
-        f:close()
-        print("To make dkim signing working, you need to place the following record in your DNS zone:")
-        print(content)
+        local rsa = require "rspamd_rsa"
+        local sk, pk = rsa.keypair(2048)
+        sk:save(privkey_file, 'pem')
+        print("You need to chown private key file to rspamd user!!")
+        print("To make dkim signing working, to place the following record in your DNS zone:")
+        print_public_key(tostring(pk))
```

Comment:
Why do we need `tostring` here?

Reply:
Generally speaking, function `print_public_key()` is slightly modified function from dkim_keygen.lua, and when you don't use `tostring`, function simply throws an error. I think because `pk` is an object and later on in `public_print_key()` its treated like a string by function `rspamd_util.encode_base64(pk)`

Reply:
`rspamd_util.encode_base64` should clearly accept `rspamd_text` as well as a `string`.

Reply:
Ah, but `pk` is not a text, it is another userdata type. I see then.